src/concrete/ml/sklearn/base.py

```diff
@@ -1690,3 +1697,358 @@ def predict_proba(self, X: Data, fhe: Union[FheMode, str] = FheMode.DISABLE) ->
         y_logits = self.decision_function(X, fhe=fhe)
         y_proba = self.post_processing(y_logits)
         return y_proba
+
+
+# pylint: disable=invalid-name,too-many-instance-attributes
+class SklearnKNeighborsMixin(BaseEstimator, sklearn.base.BaseEstimator, ABC):
+    """A Mixin class for sklearn KNeighbors models with FHE.
+
+    This class inherits from sklearn.base.BaseEstimator in order to have access to scikit-learn's
+    `get_params` and `set_params` methods.
+    """
+
+    def __init_subclass__(cls):
+        for klass in cls.__mro__:
+            # pylint: disable-next=protected-access
+            if getattr(klass, "_is_a_public_cml_model", False):
+                _NEIGHBORS_MODELS.add(cls)
+                _ALL_SKLEARN_MODELS.add(cls)
+
+    def __init__(self, n_bits: Union[int, Dict[str, int]] = 3):
+        """Initialize the FHE knn model.
+
+        Args:
+            n_bits (int, Dict[str, int]): Number of bits to quantize the model. If an int is passed
+                for n_bits, the value will be used for quantizing inputs and weights. If a dict is
+                passed, then it should contain "op_inputs" and "op_weights" as keys with
+                corresponding number of quantization bits so that:
+                    - op_inputs : number of bits to quantize the input values
+                    - op_weights: number of bits to quantize the learned parameters
+                Default to 3.
+        """
+        self.n_bits: Union[int, Dict[str, int]] = n_bits
+
+        #: The quantizer to use for quantizing the model's weights
+        self._weight_quantizer: Optional[UniformQuantizer] = None
+        self._q_X_fit_quantizer: Optional[UniformQuantizer] = None
+        self._q_X_fit: numpy.ndarray
```

Comment:
could you add a small comment explaining what are these attributes / why are they necessary ?

Reply:
with KNN, you don’t need a training phase at all, we do not talk a bout weights. Instead we we keep in memory the train base to calculate distances and similarities, commonly called _X_fit (from scikit learn).

Added a comment

Reply:
yes I meant in the code so thanks ! however, it's better to write comments above each attributes starting with "x:" (for doc visualization) , for both `_q_X_fit_quantizer` and `_q_X_fit`

Reply:
And now ?